### ISSUE TITLE
Incorrect Status (date) of transactions

### ISSUE THREAD

**Issue by HODL2MOON** [cryptoadvance/specter-desktop#466]
I imported an existing ledger account (nested segwit and account 1 instead of 0) that was already holding funds and transactions. After rescanning for UTXO and a full blockchain rescan the timestamps for the different transactions is incorrect. It seems the number of confirmations is correct. Over 30 different transactions give 3 the same dates. Eg 10k and 140k confirmations both show 27 july 2020.

**Comment by ChineseGeek** [comment]
Same issue here with an imported ColdCard wallet that already had coins.

**Comment by ben-kaufman** [comment]
I just experienced this issue myself. Seems like it's a bug on Bitcoin Core which is super surprising but I queried the RPC directly and it returns wrong result. This doesn't happen all the time though, which is weird, but seems like sometimes when doing a rescan Bitcoin Core set the time of all transactions it found to the time of the last transaction. When using `txindex=1`, doing `getrawtransaction` returns the transaction with the correct time, but still the wallet's `gettransaction` and `listtransaction` commands return wrong time. I will open an issue on Bitcoin Core to see if this is a known problem...
```
bitcoin-cli --testnet -rpcwallet=wallet getrawtransaction 8bacfc9c63d181e9d303f57f550fe72870afd640c961466829c6212bb23a72fe 1
{
  "txid": "8bacfc9c63d181e9d303f57f550fe72870afd640c961466829c6212bb23a72fe",
  "hash": "3de22064b8fec52fb183a8ad46fd011e6c6cab9906a008afa3d649d11bd250f1",
  "version": 2,
  "size": 222,
  "vsize": 141,
  "weight": 561,
  "locktime": 0,
  "vin": [
    {
      "txid": "685571776adce2ac3332faf6ed125235366cc3cb464604c9234bf371f1ab4e02",
      "vout": 0,
      "scriptSig": {
        "asm": "",
        "hex": ""
      },
      "txinwitness": [
        "30440220516d0b313aae44f639a111dac8f27bc1fac446345aaad51852a784620ac62c690220433cb7fdeb78cd84574ec0ddd08789a248fe19b71759e1b3b448ff9c3f980be401",
        "03eca8ceb44c4d930ab89fb5d721d02a745e4aeb0f75993fa535e98fb6c7f6e16e"
      ],
      "sequence": 4294967294
    }
  ],
  "vout": [
    {
      "value": 0.53891045,
      "n": 0,
      "scriptPubKey": {
        "asm": "0 f3020d45a9ccf870ddf4176fde4420931ffd9af6",
        "hex": "0014f3020d45a9ccf870ddf4176fde4420931ffd9af6",
        "reqSigs": 1,
        "type": "witness_v0_keyhash",
        "addresses": [
          "tb1q7vpq63dfenu8ph05zahau3pqjv0lmxhk73dsjm"
        ]
      }
    },
    {
      "value": 0.10000000,
      "n": 1,
      "scriptPubKey": {
        "asm": "0 bf702bf755da860153e7b1043fea160c6dcd9f1a",
        "hex": "0014bf702bf755da860153e7b1043fea160c6dcd9f1a",
        "reqSigs": 1,
        "type": "witness_v0_keyhash",
        "addresses": [
          "tb1qhaczha64m2rqz5l8kyzrl6skp3kum8c6m0c2h4"
        ]
      }
    }
  ],
  "hex": "02000000000101024eabf171f34b23c9044646cbc36c36355212edf6fa3233ace2dc6a777155680000000000feffffff02e54f360300000000160014f3020d45a9ccf870ddf4176fde4420931ffd9af68096980000000000160014bf702bf755da860153e7b1043fea160c6dcd9f1a024730440220516d0b313aae44f639a111dac8f27bc1fac446345aaad51852a784620ac62c690220433cb7fdeb78cd84574ec0ddd08789a248fe19b71759e1b3b448ff9c3f980be4012103eca8ceb44c4d930ab89fb5d721d02a745e4aeb0f75993fa535e98fb6c7f6e16e00000000",
  "blockhash": "00000000000000bc3b97cbe323c4695f39a65cc1d73060dca8ad084fb0ce61b5",
  "confirmations": 27528,
  "time": 1600623861,
  "blocktime": 1600623861
}
```
```
bitcoin-cli --testnet -rpcwallet=wallet gettransaction 8bacfc9c63d181e9d303f57f550fe72870afd640c961466829c6212bb23a72fe
{
  "amount": -0.10000000,
  "fee": -0.00000142,
  "confirmations": 27528,
  "blockhash": "00000000000000bc3b97cbe323c4695f39a65cc1d73060dca8ad084fb0ce61b5",
  "blockheight": 1835261,
  "blockindex": 24,
  "blocktime": 1600623861,
  "txid": "8bacfc9c63d181e9d303f57f550fe72870afd640c961466829c6212bb23a72fe",
  "walletconflicts": [
  ],
  "time": 1601486746,
  "timereceived": 1602570577,
  "bip125-replaceable": "no",
  "details": [
    {
      "involvesWatchonly": true,
      "address": "tb1q7vpq63dfenu8ph05zahau3pqjv0lmxhk73dsjm",
      "category": "send",
      "amount": -0.53891045,
      "label": "Change #5",
      "vout": 0,
      "fee": -0.00000142,
      "abandoned": false
    },
    {
      "involvesWatchonly": true,
      "address": "tb1qhaczha64m2rqz5l8kyzrl6skp3kum8c6m0c2h4",
      "category": "send",
      "amount": -0.10000000,
      "vout": 1,
      "fee": -0.00000142,
      "abandoned": false
    },
    {
      "involvesWatchonly": true,
      "address": "tb1q7vpq63dfenu8ph05zahau3pqjv0lmxhk73dsjm",
      "category": "receive",
      "amount": 0.53891045,
      "label": "Change #5",
      "vout": 0
    }
  ],
  "hex": "02000000000101024eabf171f34b23c9044646cbc36c36355212edf6fa3233ace2dc6a777155680000000000feffffff02e54f360300000000160014f3020d45a9ccf870ddf4176fde4420931ffd9af68096980000000000160014bf702bf755da860153e7b1043fea160c6dcd9f1a024730440220516d0b313aae44f639a111dac8f27bc1fac446345aaad51852a784620ac62c690220433cb7fdeb78cd84574ec0ddd08789a248fe19b71759e1b3b448ff9c3f980be4012103eca8ceb44c4d930ab89fb5d721d02a745e4aeb0f75993fa535e98fb6c7f6e16e00000000"
}
```

**Comment by ben-kaufman** [comment]
Opened https://github.com/bitcoin/bitcoin/issues/20181 , will follow up here based on that...

**Comment by MyNameIsOka** [comment]
Sorry for writing to a month old thread but I wanted to mention that I experienced the same problem.
After doing a rescan of an imported Ledger which holds funds, all transactions show the same date.
When I search for the txIds in my local block explorer running on myNodeBTC, it shows the correct date.
Could it be that Specter Wallet is parsing the dates incorrectly?

**Comment by BitcoinTsunami** [comment]
Hi I've proposed a patch for Bitcoin Core (https://github.com/bitcoin/bitcoin/issues/20181 thread)

From what I've seen, you can't correct date after bad transaction date determination.

If you dont want to wait for an update (It will be long because event if it's a Bitcoin Core bug, Bitcoin Core is unaffected without API interraction from external tools, like specter) :
- patch Bitcoin Core with my patch and run it
- backup you specter wallet
- destroy your specter wallet
- import your backup.
- revert to a standard Bitcoin Core version.

Important : you don't need to keep up the patched version. Incorrect date determination is only due to the rescanning process.

**Comment by ben-kaufman** [comment]
Closing as this should now be resolved by #738 and also most updates regarding this are done on #680 which is essentially a more recent duplicate, so if it's still an issue better comment over there.
